test/e2e/l7networkpolicy_test.go

```diff
@@ -429,17 +435,34 @@ type L7LogAlertEntry struct {
 }
 
 type L7LogEntry struct {
-	EventType   string           `json:"event_type"`
-	DestIP      string           `json:"dest_ip"`
-	DestPort    int32            `json:"dest_port"`
-	Protocol    string           `json:"proto"`
-	AppProtocol string           `json:"app_proto,omitempty"`
-	Http        *L7LogHttpEntry  `json:"http,omitempty"`
-	Alert       *L7LogAlertEntry `json:"alert,omitempty"`
+	EventType           string           `json:"event_type"`
+	DestIP              string           `json:"dest_ip"`
+	DestPort            int32            `json:"dest_port"`
+	Protocol            string           `json:"proto"`
+	AppProtocol         string           `json:"app_proto,omitempty"`
+	PacketBytes         []byte           `json:"packet,omitempty"`
+	Http                *L7LogHttpEntry  `json:"http,omitempty"`
+	Alert               *L7LogAlertEntry `json:"alert,omitempty"`
+	expectedPacketRegex *regexp.Regexp
 }
 
-func (e *L7LogEntry) Equal(x *L7LogEntry) bool {
-	return reflect.DeepEqual(e, x)
+// Matches the 2 L7LogEntries. If an L7LogEntry includes an expectedPacketRegex, the
+// PacketBytes field in the other L7LogEntry must match the regex. If none of the
+// L7LogEntries include an expectedPacketRegex, the PacketBytes fields must be
+// strictly equal for both entries.
+func (e *L7LogEntry) Matches(x *L7LogEntry) bool {
+	packetMatch := func(e, x *L7LogEntry) bool {
+		if e.expectedPacketRegex != nil && !e.expectedPacketRegex.MatchString(string(x.PacketBytes)) ||
+			x.expectedPacketRegex != nil && !x.expectedPacketRegex.MatchString(string(e.PacketBytes)) ||
+			x.expectedPacketRegex == nil && e.expectedPacketRegex == nil && !bytes.Equal(e.PacketBytes, x.PacketBytes) {
+			return false
+		}
+		return true
```

Comment:
I think you can use `Match` instead of `MatchString` and avoid the string conversion?

Yes this is the code I had in mind, but I find it more readable with multiple `if` statements.

Reply:
Understood, I'm still a bit curious on whether we need to compare `PacketBytes` at all, if both `L7LogEntry` contain `expectedPacketRegex`. Because in current case it would fail, as `"" PacketBytes` would not match the regex.

Reply:
If both entries contain `expectedPacketRegex`, then I think we only need to match `PacketBytes` against the `expectedPacketRegex` of the other entry.

Also in our current test code, I don't think this is an actual case (both entries containing `expectedPacketRegex`)? But regardless, the idea is to have a meaningful `Match` implementation here, and have `Match(a,b)` always be equivalent to `Match(b,a)`.

Reply:
Thanks for explaining "have `Match(a,b)` always be equivalent to `Match(b,a)`"!